test/parallel/test-dgram-multicast-set-interface.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const dgram = require('dgram');
+
+{
+  const socket = dgram.createSocket('udp4');
+
+  socket.bind(0);
+  socket.on('listening', common.mustCall(() => {
+
+    // Explicitly request default system selection
+    socket.setMulticastInterface('0.0.0.0');
+
+    socket.close();
+  }));
+}
+
+{
+  const socket = dgram.createSocket('udp4');
+
+  socket.bind(0);
+  socket.on('listening', common.mustCall(() => {
+
+    socket.close(common.mustCall(() => {
+      assert.throws(() => { socket.setMulticastInterface('0.0.0.0'); },
+                    /Not running/);
+    }));
+  }));
+}
+
+{
+  const socket = dgram.createSocket('udp4');
+
+  socket.bind(0);
+  socket.on('listening', common.mustCall(() => {
+
+    // Try to set with an invalid interfaceAddress (wrong address class)
+    assert.throws(common.mustCall(() => {
```

Comment:
You don't need `common.mustCall()` around functions passed to `assert.throws()`. If the function isn't called, then it won't throw, and the assertion will fail. `common.mustCall()` is primarily used to ensure that asynchronous callback functions are actually called.